### PR TITLE
perf: make sure we use multiple threads when scanning

### PR DESCRIPTION
**Pull request by wjones127**

### CODE REVIEW

File: rust/lance-core/src/encodings/plain.rs [hunk 1]
```diff
@@ -396,6 +396,10 @@ fn make_chunked_requests(
     //    indices[i] * byte_width + block_size < indices[i+1] * byte_width
     // It might allow slightly larger sequential reads.
     for i in 0..indices.len() - 1 {
+        // If contiguous, continue
+        if indices[i + 1] == indices[i] {
+            continue;
+        }
```
Comment (wjones127) [inline rust/lance-core/src/encodings/plain.rs:402]: This fixes a performance bug for large vectors. If we have vectors larger than the block size (1536-dim f32 vectors are 6,144 bytes while the default block size on local fs is 4K), then without this line we make a range request for **every** vector we take. 😱

File: rust/lance/src/io/exec/pushdown_scan.rs [hunk 2]
```diff
@@ -294,7 +294,12 @@ impl FragmentScanner {
         let stream = futures::stream::iter(simplified_predicates.into_iter().enumerate()).map(
             move |(batch_id, predicate)| {
                 let scanner_ref = scanner.clone();
-                async move { scanner_ref.read_batch(batch_id, predicate).await }
+                tokio::task::spawn(async move { scanner_ref.read_batch(batch_id, predicate).await })
```
Comment (westonpace) [inline rust/lance/src/io/exec/pushdown_scan.rs:297]: Would `buffered` / `buffer_unordered` not work here?

**Review by westonpace — APPROVED** [review]
Just the one question

File: rust/lance/src/io/exec/pushdown_scan.rs (hunk 2 above)
Comment (wjones127) [reply rust/lance/src/io/exec/pushdown_scan.rs:297]: We do call it below, it's just not sufficient. Each of the `read_batch()` tasks has some CPU-bound work, which can block IO of other tasks if they are in the same thread. By spawning, we distribute the tasks amongst threads, ensuring we get concurrency.

**Pull request merged**